### PR TITLE
BFB-117 - Death Fade Out

### PR DESCRIPTION
**Pull request by unoctium1**
**JIRA Ticket**
https://perigongames.atlassian.net/browse/BFB-117

**Description**
Add dissolve fade out integrated with death animation

**Screenshots / GIFs**
https://youtu.be/s5hsNINSdEw

### CODE REVIEW

File: Assets/_BForBoss/_Entities/Scripts/DummyTargetBehaviour.cs [hunk 1]
```diff
@@ -9,22 +8,29 @@ public class DummyTargetBehaviour : LifeCycleBehaviour
...
             base.Awake();
             _animator = GetComponentInChildren<Animator>();
+            _renderer = GetComponentInChildren<Renderer>();
         }
 
         protected override void LifeCycleFinished()
         {
             _animator.SetBool(DEATH_ID, true);
-            Destroy(gameObject, DEATH_ANIMATION_DURATION);
+            var tween = _renderer.material.DOFloat(MAX_DISSOLVE, DISSOLVE_ID, DEATH_ANIMATION_DURATION * 2);
```
Comment (m4rtint) [inline Assets/_BForBoss/_Entities/Scripts/DummyTargetBehaviour.cs:31]: Why is there a * 2 here?

**Comment by m4rtint** [comment]
Can you change the health property to be 2, instead of 50. So it's easier to test.

File: Assets/_BForBoss/_Entities/Scripts/DummyTargetBehaviour.cs (hunk 1 above)
Comment (unoctium1) [reply Assets/_BForBoss/_Entities/Scripts/DummyTargetBehaviour.cs:31]: Setting it to the same time as the animation, I found it faded out a little too fast and you couldn't really see the whole animation

Comment (unoctium1) [reply Assets/_BForBoss/_Entities/Scripts/DummyTargetBehaviour.cs:31]: I could expose that, probably a better idea than hardcoding it

Comment (m4rtint) [reply Assets/_BForBoss/_Entities/Scripts/DummyTargetBehaviour.cs:31]: you can name it `DISSOLVE_ANIMATION_DURATION`,  and just set it to the time of your liking.  We can think about the Tween duration separate from the death anim duration.

Comment (m4rtint) [reply Assets/_BForBoss/_Entities/Scripts/DummyTargetBehaviour.cs:31]: Like it's good to time it with the death, but the value you're using will still belong to the Dissolve tween

**Review by m4rtint — APPROVED** [review]

**Pull request merged**